src/ApiCheck/ApiListingComparer.cs

```diff
@@ -70,6 +80,130 @@ public ApiComparisonResult GetDifferences()
+            {
+                return false;
+            }
+
+            var oldSet = oldGenericParameters.OrderBy(ogp => ogp.ParameterPosition).ToArray();
+            var newSet = newGenericParameters.OrderBy(ogp => ogp.ParameterPosition).ToArray();
+            for (var i = 0; i < oldSet.Length; i++)
+            {
+                var oldParameter = oldSet[i];
+                var newParameter = newSet[i];
+                var areCompatible = AreCompatible(oldParameter, newParameter);
+                if (!areCompatible)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool AreCompatible(GenericParameterDescriptor oldParameter, GenericParameterDescriptor newParameter)
+        {
+            return ((newParameter.New && oldParameter.New) || !newParameter.New) &&
+                ((newParameter.Class && oldParameter.Class) || !newParameter.Class) &&
+                ((newParameter.Struct && oldParameter.Struct) || !newParameter.Struct) &&
+                newParameter.BaseTypeOrInterfaces.Count == oldParameter.BaseTypeOrInterfaces.Count &&
+                newParameter.BaseTypeOrInterfaces.All(btoi => oldParameter.BaseTypeOrInterfaces.Contains(btoi));
+        }
+
+        private bool HasCompatibleSetOfInterfaces(TypeDescriptor oldType, TypeDescriptor newType)
+        {
+            // An interface can't require new implemented interfaces unless they are marker interfaces (they don't have any member)
+            var newInterfaces = newType.Source.ImplementedInterfaces
+                .Where(i => !oldType.ImplementedInterfaces.Contains(TypeDescriptor.GetTypeNameFor(i.GetTypeInfo())));
+
+            return newInterfaces.All(ni => ni.GetTypeInfo().GetMembers().Length == 0);
+        }
+
+        private bool ImplementsAllInterfaces(TypeDescriptor oldType, TypeDescriptor newType)
+        {
+            var oldInterfaces = oldType.ImplementedInterfaces;
+            var newInterfaces = newType.Source.ImplementedInterfaces.Select(i => TypeDescriptor.GetTypeNameFor(i.GetTypeInfo()));
+
+            return oldInterfaces.All(oi => newInterfaces.Contains(oi));
+        }
+
+        private bool HasCompatibleVisibility(TypeDescriptor oldType, TypeDescriptor newType)
+        {
+            switch (oldType.Visibility)
+            {
+                case ApiElementVisibility.Public:
+                    return newType.Visibility == ApiElementVisibility.Public;
+                case ApiElementVisibility.Protected:
+                    // Is going from protected to public a breaking change ?
+                    return true;
+                default:
+                    throw new InvalidOperationException("Unrecognized visibility");
+            }
+        }
+
         private BreakingChange GetInvalidChange(TypeDescriptor type, MemberDescriptor member, List<ApiChangeExclusion> exclusions)
```

Comment:
Mentioned in a previous iteration but not yet addressed: `Need to come up with a better name for this... it's not "invalid".`